edc-extensions/agreements/retirement-evaluation-spi/src/main/java/org/eclipse/tractusx/edc/agreements/retirement/spi/types/AgreementsRetirementEntry.java

```diff
@@ -0,0 +1,102 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+package org.eclipse.tractusx.edc.agreements.retirement.spi.types;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import org.eclipse.edc.spi.entity.Entity;
+
+import java.time.Instant;
+
+import static java.util.Objects.requireNonNull;
+import static org.eclipse.edc.spi.constants.CoreConstants.EDC_NAMESPACE;
+import static org.eclipse.tractusx.edc.edr.spi.CoreConstants.TX_NAMESPACE;
+
+
+public class AgreementsRetirementEntry extends Entity {
+
+    public static final String AR_ENTRY_TYPE = EDC_NAMESPACE + "AgreementsRetirementEntry";
+
+    public static final String AR_ENTRY_AGREEMENT_ID = EDC_NAMESPACE + "agreementId";
+    public static final String AR_ENTRY_REASON = TX_NAMESPACE + "reason";
+    public static final String AR_ENTRY_RETIREMENT_DATE = TX_NAMESPACE + "agreementRetirementDate";
+
+    private String agreementId;
+    private String reason;
+    private String agreementRetirementDate;
+
+    public AgreementsRetirementEntry() {}
+
+    public String getAgreementId() {
+        return agreementId;
+    }
+
+    public String getReason() {
+        return reason;
+    }
+
+    public String getAgreementRetirementDate() {
+        return agreementRetirementDate;
+    }
+
+    public static class Builder extends Entity.Builder<AgreementsRetirementEntry, AgreementsRetirementEntry.Builder> {
+
+        private Builder() {
+            super(new AgreementsRetirementEntry());
+        }
+
+        @JsonCreator
+        public static Builder newInstance() {
+            return new Builder();
+        }
+
+        public Builder withAgreementId(String agreementId) {
+            this.entity.agreementId = agreementId;
+            return this;
+        }
+
+        public Builder withReason(String reason) {
+            this.entity.reason = reason;
+            return this;
+        }
+
+        public Builder withAgreementRetirementDate(String agreementRetirementDate) {
+            this.entity.agreementRetirementDate = agreementRetirementDate;
+            return this;
+        }
+
+        @Override
+        public Builder self() {
+            return this;
+        }
+
+        @Override
+        public AgreementsRetirementEntry build() {
+            super.build();
+            requireNonNull(entity.agreementId, AR_ENTRY_AGREEMENT_ID);
+            requireNonNull(entity.reason, AR_ENTRY_REASON);
+
+            if (entity.agreementRetirementDate == null) {
+                entity.agreementRetirementDate = Instant.now().toString();
```

Comment:
shouldn't we already set here as a timestamp in accordance with the decision record?

Reply:
Agree, this should be either a `long` or an `Instant` object.

Reply:
You are right, thanks for making a point out of it. It's changed in: [46ef1b7](https://github.com/eclipse-tractusx/tractusx-edc/pull/1615/commits/46ef1b7c404e70567478a983fe14a6f11c7208f8)